src/toil/job.py

```diff
@@ -66,6 +66,10 @@ def __init__(self, promisingJob, recipientJob=None):
             super().__init__(f"Job {promisingJob.description} cannot promise its return value to non-successor {recipientJob.description}")
 
 
+class ConflictingPredecessorError(Exception):
+    pass
```

Comment:
This should have a description; I think we get it when a job is already a child or follow-on of another job, and we try to add it as the other one.